saleor/graphql/checkout/mutations/checkout_delivery_method_update.py

```diff
@@ -231,12 +231,16 @@ def _update_delivery_method(
         )
         checkout.save(
             update_fields=[
-                "private_metadata",
                 "shipping_method",
                 "collection_point",
             ]
             + invalidate_prices_updated_fields
         )
+        checkout.metadata_storage.save(
```

Comment:
Will it work in case of missing metadata_storage for a given checkout? (I am asking because I saw a helper method that creates the metadata object in the fly)

Reply:
Yes, because to update or delete metadata it uses helper methods `set_external_shipping_id` and `delete_external_shipping_id` that will create `checkout.metadata_storage` if it is missing.